src/components/modules/modificationsObserver.ts

```diff
@@ -29,11 +98,45 @@ export default class ModificationsObserver extends Module {
    *
    * @param event - some of our custom change events
    */
-  public onChange(event: CustomEvent): void {
+  private particularBlockChanged(event: BlockMutationEvent): void {
     if (this.disabled || !_.isFunction(this.config.onChange)) {
       return;
     }
 
-    this.config.onChange(this.Editor.API.methods, event);
+    this.batchingOnChangeQueue.set(`block:${event.detail.target.id}:event:${event.type as BlockMutationType}`, event);
```

Comment:
Why would you need to cast here, isn't BlockMutationEvent has `BlockMutationType` as type for `type` prop?

Reply:
no, since `event` is a `CustomEvent` here where `type` is a `string`. I've [tried](https://github.com/codex-team/editor.js/pull/2349/commits/ce6299e72c66a4ca0d6f4255c74dba17b9b05461) to implement CustomEvent extensions, but it was much more over-engineered way with breaking API change (onChange handler accepts not `CustomEvent`, but custom classes), so I've [got rid](https://github.com/codex-team/editor.js/pull/2349/commits/f298a088907ae2a2a3628cb3259ad1ce35235e09) of that logic. 

So this cast is telling a compiler that `event.type` here is a known event name.

<img width="1165" alt="image" src="https://github.com/codex-team/editor.js/assets/3684889/ea529f20-52a4-46a4-9354-749e36065a67">